### DISCUSSION TITLE
Client nodes unable to put or publish data ?

### DISCUSSION THREAD

**Discussion by kaldubin** [eclipse-zenoh/roadmap#49] [Q&A] [answered]
Hello Zenohers,

I'm currently using the python API on a project and discovered that since version 0.6.0.dev0, a node in client mode cannot use the put or pub features.
There is no error messages or warning, the only thing we can see is that the subscribers nodes don't see the publication.

Is it a bug or an intended behavior ?
I didn't see any information on those and the examples are still giving the choice between peer mode and client mode.

**Comment by kydos** [comment]
Hello @kaldubin, did you try running the examples shipped with the distribution? Can you enable the logging on the router by setting the variable `RUST_LOG=debug`?

I also suggest that you ping us on [Zenoh's Discord server](https://discordservers.com/server/914168414178779197)

**Comment by p-avital** [comment]
Hi @kaldubin 

I've just given the examples in master a try and they seem to work fine.

Would you mind [opening a bug report on the zenoh-python repo](https://github.com/eclipse-zenoh/zenoh-python/issues/new?assignees=&labels=bug&template=bug_report.yml&title=%5BBug%5D+) so we can figure this out ? The global project roadmap isn't really the place for bug-hunt :)

Some more precision on the `RUST_LOG=debug` environment variable which might sound confusing if you're not using Rust, the python API is just a binding over Zenoh's Rust core, which relies on a library for logging which will only log if given a level through the `RUST_LOG` environment variable.

**Comment by kaldubin** [comment, marked as answer]
Hi @kydos @p-avital 
Sorry, I thought it would be better than creating an issue as I might have misunderstood how zenoh works.

I created an [issue](https://github.com/eclipse-zenoh/zenoh-python/issues/69) in the zenoh-python repo and might join the discord server quickly in the future.
